diff --git a/miniupnpd.c b/miniupnpd.c
index 34b2eda..aba1322 100644
--- a/miniupnpd.c
+++ b/miniupnpd.c
@@ -2258,12 +2258,6 @@ main(int argc, char * * argv)
 		if (getifaddr(ext_if_name, if_addr, INET_ADDRSTRLEN, &addr, NULL) < 0) {
 			syslog(LOG_WARNING, "Cannot get IP address for ext interface %s. Network is down", ext_if_name);
 			disable_port_forwarding = 1;
-		} else if (addr_is_reserved(&addr)) {
-			syslog(LOG_INFO, "Reserved / private IP address %s on ext interface %s: Port forwarding is impossible", if_addr, ext_if_name);
-			syslog(LOG_INFO, "You are probably behind NAT, enable option ext_perform_stun=yes to detect public IP address");
-			syslog(LOG_INFO, "Or use ext_ip= / -o option to declare public IP address");
-			syslog(LOG_INFO, "Public IP address is required by UPnP/PCP/PMP protocols and clients do not work without it");
-			disable_port_forwarding = 1;
 		}
 	}
 
@@ -2550,13 +2544,7 @@ main(int argc, char * * argv)
 					disable_port_forwarding = 1;
 				} else {
 					int reserved = addr_is_reserved(&addr);
-					if (!disable_port_forwarding && reserved) {
-						syslog(LOG_INFO, "Reserved / private IP address %s on ext interface %s: Port forwarding is impossible", if_addr, ext_if_name);
-						syslog(LOG_INFO, "You are probably behind NAT, enable option ext_perform_stun=yes to detect public IP address");
-						syslog(LOG_INFO, "Or use ext_ip= / -o option to declare public IP address");
-						syslog(LOG_INFO, "Public IP address is required by UPnP/PCP/PMP protocols and clients do not work without it");
-						disable_port_forwarding = 1;
-					} else if (disable_port_forwarding && !reserved) {
+					if (disable_port_forwarding && !reserved)
 						syslog(LOG_INFO, "Public IP address %s on ext interface %s: Port forwarding is enabled", if_addr, ext_if_name);
 						disable_port_forwarding = 0;
 					}
